### DISCUSSION TITLE
component not rerendering on initial load and stuck on fallback but will with manual refresh

### DISCUSSION THREAD

**Discussion by QuocLam-io** [vercel/next.js#73706] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

```bash
Operating System:
  Platform: darwin
  Arch: arm64
  Version: Darwin Kernel Version 24.1.0: Thu Oct 10 21:03:11 PDT 2024; root:xnu-11215.41.3~2/RELEASE_ARM64_T6020
  Available memory (MB): 16384
  Available CPU cores: 10
Binaries:
  Node: 21.2.0
  npm: 10.2.3
  Yarn: N/A
  pnpm: N/A
Relevant Packages:
  next: 15.0.2 // There is a newer version (15.0.3) available, upgrade recommended! 
  eslint-config-next: 15.0.2
  react: 19.0.0-rc-02c0e824-20241028
  react-dom: 19.0.0-rc-02c0e824-20241028
  typescript: 5.6.2
Next.js Config:
  output: N/A
 ⚠ There is a newer version (15.0.3) available, upgrade recommended! 
   Please try the latest canary version (`npm install next@canary`) to confirm the issue still exists before creating a new issue.
   Read more - https://nextjs.org/docs/messages/opening-an-issue
```

### Which example does this report relate to?

with-not-sure

### What browser are you using? (if relevant)

Chrome 132.0.6834.15

### How are you deploying your application? (if relevant)

have not yet deployed

### Describe the Bug

I have a Landing Component that has a Loading Wrapper component. On the initial load, the app is stuck on the fallback but when I manually refresh, the app works as intended and checks to see if an isLoaded variable is true and renders properly
![Loading Wrapper Component](https://github.com/user-attachments/assets/17273e6a-86
![Landing Component](https://github.com/user-attachments/assets/a25cf3ed-8f55-44ba-a8f6-ebe99c62e01f)
1b-426b-b4e4-bbd54d945cf5)

### Expected Behavior

While the fallback (CarraigeLoader) is rendered, the useEffect sets different variables and then the page should rerender, showing another component instead of the loader

### To Reproduce

My repo can be cloned at: https://github.com/QuocLam-io/defunct-challenger

The bug only happens on the initial load but when you refresh, it functions as expected. To replicate, I have to disconnect the app and run npm run dev again

**Comment by QuocLam-io** [comment]
Hmm I uploaded two images but the other one isn't showing, I'll upload it here
![Loading Wrapper Component](https://github.com/user-attachments/assets/e07ac4ad-4cd7-4866-84d7-c2dc830409ae)

**Comment by icyJoseph** [comment]
The repository link you shared is 404'ing, maybe it is set to private? I found it, https://github.com/QuocLam-io/dailychallenger/blob/main/src/pages/LandingPage.tsx, but that's not a Next.js project.

**Comment by icyJoseph** [comment]
Anyway, this looks like `loadFn()`, for some reason, rejects, and then no `.catch` block is there to set the state back to not loading.

What's not clear is how do you setup your localStorage item? You are also doing `JSON.parse`, without try catch, that's bound to cause issues.

**Comment by QuocLam-io** [comment]
> Anyway, this looks like `loadFn()`, for some reason, rejects, and then no `.catch` block is there to set the state back to not loading.
> 
> What's not clear is how do you setup your localStorage item? You are also doing `JSON.parse`, without try catch, that's bound to cause issues.

Ah sorry, I changed the name of the repo. The url should work now. I encountered an interesting behavior. When I set a console.log, on the initial render, it appears in the terminal and any refreshes afterwards it'll log in the browser. Not sure if the initial SSR is the cause for this behavior.

**Comment by icyJoseph** [comment]
Just a heads-up, you are using Casing in your imports without following a proper system. 

For example you have:

```js
import Navbar from '@/Components/navbar/Navbar';
```

But your components folder is, well, `src/components`,  and `navbar` is, `src/components/Navbar/NavBar`, which work on Windows OS and MacOS, but will fail on Linux, which is where most projects get deployed to.

- https://learn.microsoft.com/en-us/windows/wsl/case-sensitivity

> Windows file system treats file and directory names as case-insensitive. FOO.txt and foo.txt will be treated as equivalent files.
> Linux file system treats file and directory names as case-sensitive. FOO.txt and foo.txt will be treated as distinct files.

So I am exploring your project on Stackblitz, which is Linux based, and everything is broken, because of the imports.

**Comment by icyJoseph** [comment]
I've patched your project, to just get the LandingPage loading, nothing else, no auth, nada: https://stackblitz.com/edit/github-wm38ohso?file=src%2FComponents%2FWrappers%2FLandingClient.tsx

And it works fine... Have you tested this in incognito mode, or cleaning your local storage, page data altogether?

Could you record a video of the issue happening?
